src/TorchSharp/TorchVision/Image.cs

```diff
@@ -0,0 +1,177 @@
+using System.IO;
+using System.Threading.Tasks;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.torchvision
+{
+    public static partial class io
+    {
+        /// <summary>
+        /// <cref>Imager</cref> to be used when a <cref>torchvision.io</cref> image method's <c>imager</c> is unspecified.
+        /// </summary>
+        public static Imager DefaultImager { get; set; }
+
+        /// <summary>
+        /// Abstract class providing a generic way to decode and encode images as <cref>Tensor</cref>s.
+        /// Used by <cref>torchvision.io</cref> image methods.
+        /// </summary>
+        public abstract class Imager
+        {
+            /// <summary>
+            /// Reads the contents of an image file and return the image's file format.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <returns>The detected image format.</returns>
+            public abstract ImageFormat DetectFormat(byte[] image);
+            /// <summary>
+            /// Reads the contents of an image file and returns the result as a <cref>Tensor</cref>.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <param name="mode">Image read mode.</param>
+            /// <remarks>
+            /// The image format is detected from image file contents.
+            /// </remarks>
+            /// <returns>
+            /// <cref>Tensor</cref> with <c>shape = [color_channels, image_height, image_width]</c> and <c>dtype = uint8</c>.
+            /// </returns>
+            public Tensor DecodeImage(byte[] image, ImageReadMode mode = ImageReadMode.UNCHANGED)
+            {
+                return DecodeImage(image, DetectFormat(image), mode);
+            }
+            /// <summary>
+            /// Reads the contents of an image file and returns the result as a <cref>Tensor</cref>.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <param name="format">Image format.</param>
+            /// <param name="mode">Image read mode.</param>
+            /// <returns>
+            /// <cref>Tensor</cref> with <c>shape = [color_channels, image_height, image_width]</c> and <c>dtype = uint8</c>.
+            /// </returns>
+            public abstract Tensor DecodeImage(byte[] image, ImageFormat format, ImageReadMode mode = ImageReadMode.UNCHANGED);
+            /// <summary>
+            /// Encodes a <cref>Tensor</cref> with <c>shape = [color_channels, image_height, image_width]</c> into an array of bytes.
+            /// </summary>
+            /// <param name="image"><cref>Tensor</cref> with <c>shape = [color_channels, image_height, image_width]</c>.</param>
+            /// <param name="format">Image format.</param>
+            /// <returns>The encoded image.</returns>
+            public abstract byte[] EncodeImage(Tensor image, ImageFormat format);
```

Comment:
Maybe add an EncodeImage that takes a stream?

Reply:
Or did you mean an `EncodeImage` image that takes a stream to write to and doesn't return a byte array?

Reply:
Sorry, yes, that's what I meant -- convert directly from a Tensor into a stream.